spring-batch-core/src/main/java/org/springframework/batch/core/jsr/step/PartitionStep.java

```diff
@@ -102,6 +103,7 @@ protected void doExecute(StepExecution stepExecution) throws Exception {
 	/* (non-Javadoc)
 	 * @see org.springframework.batch.core.step.StepLocator#getStep(java.lang.String)
 	 */
+	@Nullable
```

Comment:
This `@Nullable` here does not make sense as the implementation throws `NoSuchStepException` when no step is found with the given name. The issue is that the behaviour is not consistent across the framework: some implementations return `null` (like  [SimpleJob](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/job/SimpleJob.java#L117)) and some others throw `NoSuchStepException` like in here. I believe it is the contract `StepLocator#getStep` that should be fixed to throw an exception instead of returning null (in order to make it consistent with [`JobLocator`](https://github.com/spring-projects/spring-batch/blob/master/spring-batch-core/src/main/java/org/springframework/batch/core/configuration/JobLocator.java#L42)), but my concern is that it would be a breaking change.

Reply:
It looks like the contract `StepLocator` has been changed incorrectly [here](https://github.com/spring-projects/spring-batch/commit/276acd1960b0e2578596a0d3507ad624ace9037c). I reverted that change and adapted the PR accordingly.

Reply:
Yes, you are right. Thanks for the changes.